lumicks/pylake/detail/image.py

```diff
@@ -114,8 +114,11 @@ def line_timestamps_image(time_stamps, infowave, pixels_per_line):
     pixels_per_line : int
         The number of pixels on the fast axis of the scan.
     """
-    min_stamps = reconstruct_image(time_stamps, infowave, shape=(pixels_per_line,), reduce=np.min)
-    return min_stamps.astype(np.int64)[:, 0]
+    infowave, valid_idx = discard_zeros(infowave)
+    time_stamps = time_stamps[valid_idx]
+    pixel_start_idx = np.flatnonzero(infowave == InfowaveCode.pixel_boundary)
+    pixel_start_idx = np.concatenate(([0], pixel_start_idx + 1))
+    return time_stamps[pixel_start_idx[0:-1:pixels_per_line]]
```

Comment:
I am not sure I can parse this ... slices everything but the last element, with a step size of `pixels_per_line` (stride). And thats the indices we extract from `time_stamps` array?

Reply:
Yes.

We find the pixel boundaries. We go to the DAQ sample immediately after the pixel boundary. We want the line starts, so we want the ones that happen every `pixels_per_line`. The last one is dropped since it would be beyond the end of the time_stamp range (last pixel of a line plus 1).